tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -364,123 +374,91 @@ ROOT::Experimental::RNTupleJoinProcessor::RNTupleJoinProcessor(const RNTupleOpen
       auto &field = value.GetField();
       const auto &fieldName = field.GetQualifiedFieldName();
 
+      auto isAuxParent = std::find_if(auxNTuples.cbegin(), auxNTuples.cend(), [&fieldName](const RNTupleOpenSpec &n) {
```

Comment:
I think this part could use a comment to explain why we want to skip the loop when this condition is true...it's not immediately obvious to me.